### PR TITLE
[FIX] website_sale_coupon: Change parameters for method "cart"

### PR DESCRIPTION
**Pull request by axelpriem**
The original method in website_sale is defined as:
def cart(self, access_token=None, revive='', **post):

So we must include parameters access_token and revive here. Otherwise we might get an error if another module also inherits the same method (with the correct parameters). If the method from website_sale_coupon is processed first, one parameter (post) is expected. So when another module tries to pass three parameters, an error will appear.

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by fmdl** [comment]
cc @nim-odoo

File: addons/website_sale_coupon/controllers/main.py [hunk 1]
```diff
@@ -23,10 +23,10 @@ def payment(self, **post):
         return super(WebsiteSale, self).payment(**post)
 
     @http.route(['/shop/cart'], type='http', auth="public", website=True)
-    def cart(self, **post):
+    def cart(self, access_token=None, revive='', **post):
         order = request.website.sale_get_order()
         order.recompute_coupon_lines()
-        return super(WebsiteSale, self).cart(**post)
+        return super(WebsiteSale, self).cart(access_token, revive, **post)
```
Comment (JKE-be) [inline addons/website_sale_coupon/controllers/main.py:29]:
```suggestion
        return super(WebsiteSale, self).cart(access_token=access_token, revive=revive, **post)
```

**Review by JKE-be — COMMENTED** [review]
Thanks for fix, can you sign cla and ammend your commit message to follow our guidelines

[FIX] website_sale_coupon: ...

File: addons/website_sale_coupon/controllers/main.py (hunk 1 above)
Comment (axelpriem) [reply addons/website_sale_coupon/controllers/main.py:29]: I have added the CLA md file. But the check on CLA is still failing. What else do I need to do?

Comment (JKE-be) [reply addons/website_sale_coupon/controllers/main.py:29]: when you commit with Github, it doesn't use your email and so your CLA doesn't match.
Eg: 45359880+axelpriem@users.noreply.github.com

And, could you ammend your commit message to follow guidelines : https://www.odoo.com/documentation/14.0/reference/guidelines.html#commit-message-structure

thank you

Comment (axelpriem) [reply addons/website_sale_coupon/controllers/main.py:29]: I have squashed commits and modified the message.
But I don't understand what I have to change for the CLA matching of my e-mail address. Could you explain please?

Comment (JKE-be) [reply addons/website_sale_coupon/controllers/main.py:29]: you need to commit with the same email than in your cla...

`git config --list --show-origin | grep user.email`

This email doesn't match any CLA
image

Btw it is probably a wrong idea to remove others users from cla ;) 
If they contributed in the past, they probably need to stay in the cla.
Maybe @mart-e can explain the best practice to clean cla with developer if it was really your intention (some doubts)

Comment (axelpriem) [reply addons/website_sale_coupon/controllers/main.py:29]: OK, my email address in Github was not configured correctly, I didn't know that. It's fixed now.
I have changed the CLA file so the already present developers are not removed. This was indeed NOT my intention! I checked for an existing file before, but couldn't find it. Guess I overlooked it :-)
Thank you very much for signalling this to me!
All checks have now passed :-)

Comment (JKE-be) [reply addons/website_sale_coupon/controllers/main.py:29]: You still need to amend your first commit with --author= to update it.
And while you are doing it, can you please amend your commit message for cla following guidelines ^^

> [CLA] for signing the Odoo Individual Contributor License;


thank you

Comment (JKE-be) [reply addons/website_sale_coupon/controllers/main.py:29]: And if you can apply the suggested change to make named params clear ;) 

```py
        return super(WebsiteSale, self).cart(access_token=access_token, revive=revive, **post)
```

Comment (JKE-be) [reply addons/website_sale_coupon/controllers/main.py]: Just need to squash forst and last commit now, and it hsould be ok :1st_place_medal:

**Comment by fmdl** [comment]
@JKE-be it is possible to merge ? Thx

**Comment by JKE-be** [comment]
@robodoo r+ rebase-ff

**Comment by JKE-be** [comment]
Checking the fail build, I see (not related to the fail)) that you still have the wrong diff of CLA in your first commit...

Can you please have One commit where you add / update your CLA, and one commit with the fix, and avoid to mix both.


Thank you

**Pull request closed without merging**